extensions/trust-framework-policies/src/main/java/org/eclipse/edc/trustframework/policy/seeding/PolicySeedingExtension.java

```diff
@@ -0,0 +1,95 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.policy.model.Permission;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Inject;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Setting;
+import org.eclipse.dataspaceconnector.spi.policy.engine.PolicyEngine;
+import org.eclipse.dataspaceconnector.spi.policy.engine.RuleBindingRegistry;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.system.configuration.Config;
+import org.eclipse.edc.trustframework.policy.seeding.exception.PolicyConfigFileException;
+import org.eclipse.edc.trustframework.policy.seeding.function.VerifiableCredentialEvaluationFunction;
+import org.eclipse.edc.trustframework.policy.seeding.model.PolicyEntry;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Extension for seeding {@link PolicyEngine} and {@link RuleBindingRegistry} based on a static config json file.
+ */
+public class PolicySeedingExtension implements ServiceExtension {
+    @Setting(value = "Path to the static configuration file containing the policy entries")
+    private static final String CONFIG_FILE_PATH_SETTING = "edc.trustframework.policies.file.path";
+
+    private static final String DEFAULT_CONFIG_FILE_PATH = "gaiax-policies-22_10.json";
+
+    @Inject
+    private RuleBindingRegistry ruleBindingRegistry;
+
+    @Inject
+    private PolicyEngine policyEngine;
+
+    @Override
+    public String name() {
+        return "Trust Framework Policy Seeding";
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var entries = loadEntries(context.getConfig(), context.getTypeManager().getMapper());
+        entries.forEach(entry -> {
+                    var function = new VerifiableCredentialEvaluationFunction(context.getMonitor(), context.getTypeManager().getMapper(), entry.getJsonPath());
+                    policyEngine.registerFunction(entry.getScope(), Permission.class, entry.getType(), function);
+                    ruleBindingRegistry.bind(entry.getType(), entry.getScope());
+                }
+        );
+    }
+
+
+    private List<PolicyEntry> loadEntries(Config config, ObjectMapper mapper) {
+        var tr = new TypeReference<List<PolicyEntry>>() {
+        };
+        try (var is = Optional.ofNullable(config.getString(CONFIG_FILE_PATH_SETTING, null))
+                .map(PolicySeedingExtension::fromFileSystem).orElse(fromClassPath())) {
+            return mapper.readValue(is, tr);
+        } catch (IOException e) {
+            throw new PolicyConfigFileException("Failed to load policy entries", e);
+        }
+    }
+
+    private static InputStream fromFileSystem(String path) {
```

Comment:
Why ? 
`static` keyword guarantees you are not using any class argument, which is a good feature to my point of view.

Reply:
I would avoid private static methods as much as possible as they are symptom that probably that logic needs to stay elsewhere, exception done by the logic being done before calling the `super` or `this` method (if any needed) by a constructor, that in fact cannot be done in other way.

In this specific case, looking closer at it I think this is providing a bug, because the input stream will be closed by the try-with-resource so it could not be accessed by the mapper at line 76, so it can be inlined.